### PR TITLE
perf(reporter): add option to limit result types to be processed

### LINKED ISSUE #512: Axe reporter is very slow with large datasets
On extremely large pages, it is possible for the reporter to take several times longer than the actual audit to complete. The main problem appears to be in `helpers.processAggregate()`. Running it 30 000 times took several minutes to complete. Part of the issue here is that the extension is hugely inefficient here because it's processing passes, which won't actually get displayed. 

Part of the solution should be in preventing that. But there's certainly an extra problem. The method doesn't (or at least shouldn't) do anything that's particularly slow, so we need to figure out what's so slow and we should fix it.

This problem can be reproduced on the following page: 
https://dev.office.com/fabric#/components/detailslist

### PR DESCRIPTION
**Pull request by isner**
This allows us to specify a short list of result types that we want `helper.processAggregate` to bother processing. For some reporting scenarios, we don't want to bother processing 'passes', for instance. This causes a massive performance improvement for those scenarios.

We can leverage this option in the axe extension to speed up performance there as well.

Closes https://github.com/dequelabs/axe-core/issues/512

### CODE REVIEW

**Comment by isner** [comment]
@WilcoFiers Tests added. Let me know if the documentation is insufficient.

File: doc/API.md [hunk 1]
```diff
@@ -231,7 +231,7 @@ axe.run(context, options, callback);
 #### Parameters axe.run
 
 * [`context`](#context-parameter): (optional) Defines the scope of the analysis - the part of the DOM that you would like to analyze. This will typically be the `document` or a specific selector such as class name, ID, selector, etc.
-* [`options`](#options-parameter): (optional) Set of options passed into rules or checks, temporarily modifying them. This contrasts with `axe.configure`, which is more permanent. [See below for more information](#axerun-parameters)
```
Comment (isner) [inline doc/API.md]: The documentation for `axe.configure` is _above_, not below, and the internal link is invalid. I figured we ought to simply remove this.

**Comment by dylanb** [comment]
@WilcoFiers @isner did the CircleCI build run for this PR?

**Comment by marcysutton** [comment]
@dylanb it says it did, you can find the integrations under "View details" on the merged commit above

**Pull request merged**